src/OrchardCore.Modules/OrchardCore.Media/Processing/MediaTokenService.cs

```diff
@@ -98,30 +131,97 @@ public bool TryValidateToken(IDictionary<string, string> commands, string token)
             return false;
         }
 
-        private static string CreateQueryStringTokenKey(IEnumerable<string> values)
+        /// <summary>
+        /// Specialized version with fast enumeration and no StringValues string allocation.
+        /// </summary>
+        private static string CreateQueryStringTokenKey(Dictionary<string, StringValues> values)
```

Comment:
a bit of duplication, but with concrete type overload we get fast struct enumerator instead of one wrapped via interface